src/Api/ResourceClassResolver.php

```diff
@@ -63,6 +63,39 @@ public function getResourceClass($value, string $resourceClass = null, bool $str
     /**
      * {@inheritdoc}
      */
+    public function getResourceClassFromContext($value, array $context = null, bool $strict = false): string
```

Comment:
Instead of BC break I chose to duplicate the method, I think we should deprecate the old one, let me know if you've a better idea.